validate/validate.go

```diff
@@ -335,6 +336,24 @@ func (v *Validator) CheckMounts() (msgs []string) {
 	return
 }
 
+func (v *Validator) CheckOS() (msgs []string) {
+	logrus.Debugf("check os")
+
+	if v.spec.Platform.OS != "linux" {
+		if v.spec.Linux != nil {
+			msgs = append(msgs, fmt.Sprintf("When system is not linux, Linux field should be empty"))
```

Comment:
But spec has changed in [this](https://github.com/opencontainers/runtime-spec/pull/673) PR.